SonarQube.TeamBuild.PreProcessor/TeamBuildPreProcessor.cs

```diff
@@ -42,67 +50,61 @@ public TeamBuildPreProcessor(ISonarQubeServerFactory serverFactory, ITargetsInst
             {
                 throw new ArgumentNullException("targetInstaller");
             }
+            if (analyzerInstaller == null)
+            {
+                throw new ArgumentNullException("analyzerProvider");
+            }
 
+            this.logger = logger;
             this.serverFactory = serverFactory;
             this.targetInstaller = targetInstaller;
+            this.analyzerProvider = analyzerInstaller;
         }
 
         #endregion Constructor(s)
 
         #region Public methods
 
-        public bool Execute(string[] args, ILogger logger)
+        public bool Execute(string[] args)
         {
-            if (logger == null)
-            {
-                throw new ArgumentNullException("logger");
-            }
-
             bool success;
 
-            ProcessedArgs processedArgs = ArgumentProcessor.TryProcessArgs(args, logger);
+            ProcessedArgs processedArgs = ArgumentProcessor.TryProcessArgs(args, this.logger);
```

Comment:
Most of the rest of the changes in this file are related to changing to logger to a field. I'll call out the other changes explicitly.